src/app/[locale]/client-page.tsx

```diff
@@ -0,0 +1,39 @@
+'use client';
+
+import {useQuery} from '@tanstack/react-query';
+import {SearchResult} from '@/lib/dataset-fetcher';
+import DatasetCard from '@/components/dataset-card';
+import {Dataset} from '@/lib/dataset-fetcher';
+
+interface ClientPageProps {
+  initialDatasets: SearchResult;
+  locale: string;
+}
+
+export default function ClientPage({initialDatasets, locale}: ClientPageProps) {
+  const {data: {datasets} = {}}: {data: {datasets?: Array<Dataset>}} = useQuery(
```

Comment:
1. This line looks rather complicated :) Would it be possible to not destructure the values, to make the code easier to read?
2. Would the shorter `Dataset[]` also work (instead of Array<Dataset>`)?